sonic-xcvrd/scripts/xcvrd

```diff
@@ -33,15 +33,26 @@ PLATFORM_SPECIFIC_CLASS_NAME = "SfpUtil"
 
 TRANSCEIVER_INFO_TABLE = 'TRANSCEIVER_INFO'
 TRANSCEIVER_DOM_SENSOR_TABLE = 'TRANSCEIVER_DOM_SENSOR'
+TRANSCEIVER_ERROR_TABLE = 'TRANSCEIVER_ERROR'
```

Comment:
@keboliu why not to have STATUS table instead? Then we can leverage `show interfaces transceiver` CLI group.
For example we can add `status`

Reply:
MSFT doesn't like the idea about store the SFP error status to DB and then CLI fetch the status from DB. They suggested getting the error status directly from some register, which will be more reliable and realtime, SDK will provide a new API for SFP status getting and the new CLI will get the SFP status from there. 
Actually here the new table is only for xcvrd internal use, SFP event change monitor thread will update this table when receiving an error event, DOM update process will check this table to get to know which SFP is in error status and should skip reading it's EEPROM.

Reply:
@keboliu then why we need to use caching in this case? Why not to use direct call to HW to get the latest data? This can be done before DOM update operation? Please elaborate.

Reply:
currently, we don't have a direct call(SDK API) to HW yet to get the error status.  the CLI will only be implemented after SDK provides the SDK.

Here caching SFP status is for XCVRD to check and skip reading the EEPROM because in this case, EEPROM will not be accessible, reading will cause errors. We are trying to eliminate the errors that could be caused by reading unavailable EEPROM.

Reply:
@keboliu 

> currently, we don't have a direct call(SDK API) to HW yet to get the error status. the CLI will only be implemented after SDK provides the SDK.

So the idea is to use the error code form random event to later on decide whether to read QSFP EEPROM or not? And once the SDK will provide such an API to align accordingly?

Reply:
yes.

Reply:
@keboliu ok. Thanks for clarification.